generatorreceiver/internal/generator/trace_generator.go

```diff
@@ -139,10 +139,10 @@ func (g *TraceGenerator) createSpanForServiceRouteCall(traces *pdata.Traces, ser
 	// after a sample of the current route's latency, which doesn't really
 	// make sense - but maybe it's realistic enough?
 	endTime := startTimeNanos + route.SampleLatency()
-	for s, r := range route.DownstreamCalls {
+	for _, r := range route.DownstreamCalls {
```

Comment:
nit: since we are no longer looping through a map of services (`s`) and routes (`r`) and instead looping through a slice of `Call`s, I'd recommend changing the loop variable `r` to something like `call` or `c` or `dc`